src/components/Icon/Icon.js

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+import { COMMON } from '../../utils/constants';
+import * as icons from './IconPaths';
+
+
+const IconBase = styled.svg`
+  display: inline-block;
+  vertical-align: center;
+
+  fill: ${(props) => props.fill};
+  ${COMMON};
+`;
+
+const Icon = ({
+  viewBox, type, ...props
+}) => (
+  <IconBase
+    viewBox={icons[type] ? icons[type].viewBox : '0 0 30 30'}
```

Comment:
I don't know too much about SVGs - what is this for?  is there a reason we can't make the viewBox set to this in the IconPaths file?

Reply:
viewbox is set in the <svg> element but i style it and then put the path elements inside the svg. like
```js
<svg viewbox"0 0 30 30">
 .... path 
</svg>
```

so i decided to make each element have a viewbox so I could set it. there could be another way though, but im not sure cuz i just learned svgs now haha